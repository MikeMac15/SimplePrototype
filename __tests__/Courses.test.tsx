import React from 'react';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import Courses from '@/app/(myCourses)';// Adjust the path to your component

// Mocks for AsyncStorage and other modules can be in jestSetupFile.js
jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);

describe('Courses Snapshot Test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <NavigationContainer>
          <Courses />
        </NavigationContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

