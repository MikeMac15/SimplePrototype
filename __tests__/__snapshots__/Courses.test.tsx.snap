// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Courses Snapshot Test renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  colors={
    [
      4278190080,
      4279308561,
    ]
  }
  style={
    {
      "height": "100%",
    }
  }
>
  <MockStackScreen
    options={
      {
        "headerRight": [Function],
      }
    }
  />
  <RCTScrollView>
    <View>
      <Text>
        Add a Course.
      </Text>
    </View>
  </RCTScrollView>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4284111450,
        4281545523,
      ]
    }
    style={
      {
        "alignItems": "center",
        "borderColor": "#222",
        "borderRadius": 20,
        "borderWidth": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "margin": 12,
        "marginBottom": 25,
        "padding": 5,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderRadius": 10,
          "flexDirection": "row",
          "marginVertical": 10,
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Add New Course
      </Text>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</ViewManagerAdapter_ExpoLinearGradient>
`;
